app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb

```diff
@@ -92,4 +92,39 @@ def allowed_storages(options = {})
 
     result.select { |s| s.storage_domain_type == "data" }
   end
+
+  def load_allowed_vlans(hosts, vlans)
+    ems = source_ems
+    if update_vnic_profile_supported?(ems)
+      load_v4_allowed_vlans(ems, vlans)
+    else
+      super(hosts, vlans)
+    end
+  end
+
+  def source_ems
+    src = get_source_and_targets
+    load_ar_obj(src[:ems])
+  end
+
+  def update_vnic_profile_supported?(ems)
+    ems.supports_update_vnic_profile?
+  end
+
+  def load_v4_allowed_vlans(ems, vlans)
+    ems.with_provider_connection(:version => 4) do |connection|
+      profiles = connection.system_service.vnic_profiles_service.list
```

Comment:
the connection logic should be probably in ManageIQ::Providers::Redhat::InfraManager::ApiIntegration